common/djangoapps/request_cache/__init__.py

```diff
@@ -73,11 +73,6 @@ def get_request_or_stub():
     request = crum.get_current_request()
 
     if request is None:
-        log.warning(
```

Comment:
I would also update the docstring to make it clear that this method can safely be used in non-test code, such as Celery tasks, or XBlocks.